styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -271,6 +278,8 @@ private void process(RunState state, long counterPosition, Event event,
 
     try {
       storeEvent(sequenceEvent);
+      // racy...
```

Comment:
Racy in what regard? Have we promised any invariant that this can violate in edge cases? Should be clarified, or the comment can go away.

Reply:
I think it's an old comment and still valid very likely.

Reply:
Disregard my comment. This is a new comment. Let's make it clearer why it is racy.